app/lib/saml.server.ts

```diff
@@ -0,0 +1,103 @@
+// saml server based on following PR
+// https://github.com/remix-run/examples/pull/130/files/ec66b3060fac83eec2389eb0c96aad6d8ea4aed1#diff-02d2b71e481b2495b8a72af14f09fc28238298c7f1d19a540e37c9228985b0da
+import * as samlify from 'samlify';
+import * as validator from '@authenio/samlify-node-xmllint';
+
+import logger from './logger.server';
+
+import type { IdentityProvider } from 'samlify/types/src/entity-idp';
+
+samlify.setSchemaValidator(validator);
+
+let idp: IdentityProvider;
+
+// this is needed because in development we don't want to restart
+// the server with every change, but we want to reuse the already
+// initialized idp that we created at startup, and cached on the global.
+declare global {
```

Comment:
Does this ever get used apart from dev? Could it also be included under the below conditional?

Reply:
Apparently not:

> An ambient module declaration is only allowed at the top level in a file.ts (1234)